terraform/modules/storage/blob/storage_account.tf

```diff
@@ -0,0 +1,37 @@
+module "storage_account" {
+  source = "../account"
+  name                          = var.name
+  subscription_name             = var.subscription_name
+  location                      = var.location
+  resource_group_name           = var.resource_group_name
+  account_kind                  = "BlockBlobStorage"
+  account_tier                  = "Premium"
+  account_replication_type      = var.account_replication_type
+  enable_hierarchical_namespace = var.enable_hierarchical_namespace
+  allowed_ip_addresses          = var.allowed_ip_addresses
+  allowed_subnet_ids            = var.allowed_subnet_ids
+  blob_properties               = var.blob_properties
+  tags                          = var.tags
+}
+
+
+# FIXME: https://github.com/hashicorp/terraform-provider-azurerm/issues/2977
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/resource_group_template_deployment
+resource "azurerm_resource_group_template_deployment" "container" {
```

Comment:
***Note:*** Due to a bug in `terraform` you can't deploy containers to storage accounts which implement network rules. This hack works-around that bu by using an ARM template deployment instead of a [`storage_container`](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_container) resource.